src/Export.js

```diff
@@ -34,10 +37,65 @@ const Export = ({ API, year, session }) => {
       <SiMicrosoftexchange /> Office 365
     </Dropdown.Item>
     <Dropdown.Divider />
-    <Dropdown.Item eventKey="ics" download={`${name} as of ${new Date().toLocaleDateString().replace(/(\/|\\)/g,'.')}.ics`} href={`${window.location.protocol}//${path}`}>
+    <Dropdown.Item eventKey="ics" download={`${name} as of ${new Date().toLocaleDateString().replace(/(\/|\\)/g, '.')}.ics`} href={`${window.location.protocol}//${path}`}>
       <RiCalendar2Fill /> Download ICS file
     </Dropdown.Item>
+    <Dropdown.Item as="Button">
+      <div onClick={exportToPng}>
+        <BsCardImage /> Download PNG
+      </div>
+    </Dropdown.Item>
   </DropdownButton>
 }
 
+// This function exports the current timetable to a PNG
+// This turned out being quite hacky for a number of reasons, html-to-image
+const exportToPng = () => {
+  // start screenshot at 8 am
+  const startTime = 7;
+  // end screenshot at 6pm (24 - timeFromEnd)
+  const timeFromEnd = 6;
```

Comment:
In other timezones, this might not include all the classes. We should either export all 24 hours or derive a window from the calendar's list of events that all the classes are in and export that

Reply:
Very good point, I'll look into it tonight